include/maliput/routing/route.h

```diff
@@ -231,7 +251,27 @@ class Route final {
   // Route.
   std::optional<LaneSRangeIndex> FindLaneSRangeIndex(const api::LaneSRange& lane_s_range) const;
 
+  // Finds the LaneSRangeIndex of the api::LaneSRange that is
+  // LaneSRelation::kPreceedingStraight with respect to @p lane_s_range_index
+  // api::LaneSRange.
+  //
+  // @param lane_s_range_index The index of the api::LaneSRange to find its
+  // LaneSRelation::kPreceedingStraight counterpart. It must be a valid index.
+  // @return An optional containing the LaneSRangeIndex of the api::LaneSRange.
+  std::optional<LaneSRangeIndex> FindStraightPredecessor(const LaneSRangeIndex& lane_s_range_index) const;
```

Comment:
When I wrote this method I was tempted to write a generic one:

```c++
std::optional<LaneSRangeIndex> FindLaneSRangeIndex(
    const LaneSRangeIndex& lane_s_range_index,
    const LaneSRangeRelation& relation) const;
```

That method would be better received in the public section of this class and probably useful to agents. I refrained to do so in favor of meeting the use case and focused only on the case of the kStraightPredecessor which is required to compute the LaneSRoute from any point in the Route towards the end.

Probably with this context we can determine whether it should be something that we keep at the Route level or at the Phase level. I think it will be better to keep it at the Route level.


`FindDirectionTowardsLaneSRangeWithStraightPredecessor` is a different story. That is particular method to compute the LaneSRoute. I  don't think we can make it more generic unless we convert it in any direction. That could help an agent to obtain the LaneSRoute / Phase index in arbitrary directions.

Thoughts?

Reply:
> Thoughts?

I'd keep it in Route until we have the necessity to expose some generic method as mentioned above, and then evaluate if we move it to Phase or to the public API here